corehq/apps/userreports/util.py

```diff
@@ -355,3 +369,21 @@ def unescape(value):
         # double-unescape because corehq.apps.userreports.util.get_table_name escapes twice
         return unescape(unescape(match.group(1)))
     raise ValueError(f"Expected {table_name} to start with {UCR_TABLE_PREFIX} or {LEGACY_UCR_TABLE_PREFIX}")
+
+
+def register_data_source_row_change(domain, data_source_id, row_changes):
+    from corehq.motech.repeaters.models import DataSourceRepeater
+    from corehq.motech.repeaters.signals import ucr_data_source_updated
+    try:
+        if not (
+            toggles.SUPERSET_ANALYTICS.enabled(domain)
+            and DataSourceRepeater.datasource_is_subscribed_to(domain, data_source_id)
+        ):
+            return
+
+        for row_change in row_changes:
+            update_log = DataSourceUpdateLog(domain, data_source_id=data_source_id, doc_id=row_change["doc_id"])
```

Comment:
Since you only need doc IDs you could just pass a list of doc IDs instead of `row_changes` dict.

Also you should remove duplicates (for cases where a single doc creates multiple rows).